scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -1159,13 +1163,38 @@ def _update_instance_info(self, harness_state, is_timeout):
                     self.instance.reason = "Exited with {}".format(self.returncode)
             self.instance.add_missing_case_status("blocked")
 
-    def _enqueue_char(self, stdout, queue):
+    def _enqueue_char(self, queue):
         while not self.stop_thread:
+            if not self.pipe_handle:
+                try:
+                    self.pipe_handle = win32file.CreateFile(
+                        r"\\.\pipe\\" + self.fifo_fn,
+                        win32file.GENERIC_READ | win32file.GENERIC_WRITE,
+                        0,
+                        None,
+                        win32file.OPEN_EXISTING,
+                        0,
+                        None
+                    )
+
+                    res = win32pipe.SetNamedPipeHandleState(
+                        self.pipe_handle, win32pipe.PIPE_READMODE_BYTE, None, None
+                    )
+
+                    if res == 0:
+                        continue
+
+                except win32file.error as e:
+                    if e.args[0] == 2:
+                        # Pipe is not opened yet, try again after a delay.
+                        time.sleep(1)
+
+                continue
+
             try:
-                c = stdout.read(1)
-            except ValueError:
-                # Reading on closed file exception can occur when subprocess is killed.
-                # Can be ignored.
+                c = win32file.ReadFile(self.pipe_handle, 1)[1]
+            except win32file.error:
+                # Reading on closed pipe causes an exception, e.g. if QEMU crashed. Can be ignored.
```

Comment:
Thank you for the comments. I will try to implement it using os.pipe(), but not sure, if it can work because QEMU itself creates single duplex file and twister connects to the existing pipe, while os.pipe() creates 2 new pipes itself (1 for read and 1 for write).